python/lib/sift_py/report_templates/service.py

```diff
@@ -0,0 +1,139 @@
+            rule_client_keys=[rule.client_key for rule in report_template.rules],
+            archived_date=to_timestamp_nanos(report_template.archived_date).to_pydatetime(),
+        )
+
+    def load_report_templates_from_yaml(self, paths: List[Path]) -> List[ReportTemplateConfig]:
+        report_templates = load_report_templates(paths)
+        [
+            self.create_or_update_report_template(report_template)
+            for report_template in report_templates
+        ]
+        return report_templates
+
+    def _get_report_template_by_id(self, report_template_id: str) -> Optional[ReportTemplate]:
+        req = GetReportTemplateRequest(report_template_id=report_template_id)
+        try:
+            res = cast(
+                GetReportTemplateResponse, self._report_template_service_stub.GetReportTemplate(req)
+            )
+            return cast(ReportTemplate, res.report_template) or None
+        except:
+            return None
+
+    def _get_report_template_by_client_key(self, client_key: str) -> Optional[ReportTemplate]:
+        req = GetReportTemplateRequest(client_key=client_key)
+        try:
+            res = cast(
+                GetReportTemplateResponse, self._report_template_service_stub.GetReportTemplate(req)
+            )
+            return res.report_template or None
+        except:
+            return None
+
+    def _create_report_template(self, config: ReportTemplateConfig):
+        client_keys_req = CreateReportTemplateRequestClientKeys(
+            rule_client_keys=config.rule_client_keys
+        )
+        req = CreateReportTemplateRequest(
+            name=config.name,
+            client_key=config.template_client_key,
+            description=config.description,
+            tag_names=config.tags,
+            organization_id=config.organization_id,
+            rule_client_keys=client_keys_req,
+        )
+        self._report_template_service_stub.CreateReportTemplate(req)
+
+    def _update_report_template(
+        self, config: ReportTemplateConfig, report_template: ReportTemplate
+    ):
+        tags = []
+        if config.tags:
+            tags = [ReportTemplateTag(tag_name=tag) for tag in config.tags]
+
+        archived_date = None
+        if config.archived_date:
+            archived_date = to_timestamp_pb(config.archived_date)
+
+        rules = [
+            ReportTemplateRule(client_key=client_key) for client_key in config.rule_client_keys
+        ]
+
+        updated_report_template = ReportTemplate(
+            report_template_id=report_template.report_template_id,
+            organization_id=report_template.organization_id,
+            client_key=report_template.client_key,
+            name=config.name,
+            description=config.description,
+            tags=tags,
+            rules=rules,
+            archived_date=archived_date,
+        )
+
+        field_mask = FieldMask(paths=["name", "description", "tags", "rules", "archived_date"])
```

Comment:
I was originally thinking we could be smarter about this and check each field to see which ones have actually changed, and just include those in the fieldmask, but I'm not sure if that buys us anything

Reply:
Since this is "private" that should be fine? It's only accessible via the `create_or_update_*` method so I can't see any potential footguns but I'd just double check and consider this a bit more.